src/test/java/pl/wedding/management/service/user/InMemoryRepositoryTest.java

```diff
@@ -0,0 +1,68 @@
+package pl.wedding.management.service.user;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import pl.wedding.management.service.user.model.EMail;
+import pl.wedding.management.service.user.model.User;
+
+import static org.assertj.core.api.Assertions.*;
+
+class InMemoryRepositoryTest {
+
+    private static final char[] DUMMY_PASSWORD = new char[]{'p', 'a', 's', 's', 'w', 'o', 'r', 'd'};
+    private final User constUser = createUser("someEmail@dummy.pl");
+
+    private InMemoryRepository underTest;
+
+    @BeforeEach
+    void setUp() {
+        underTest = new InMemoryRepository();
+        underTest.clearInMemoryRepository();
```

Comment:
You are creating new object. Why clean it? Do you think that it is not cleaned ?

Reply:
That's how InMemoryRepository works. Instance of HashSet is static so there is a need to clear it before each method. 
This is not the best solution but it is only temporary.

Reply:
If's static, don't create it's instance: InMemoryRepository.clearInMemoryRepository(), but that requires all methods to be static as well. Maybe internal object shouldn't be static, OR mock/in memory dba constructor should clear it instead.

Either way, you don't have to clear real database, and shouldn't have to do extra work with it's mock/in memory implementation as well. If you have to something more than with real database object, it's not good mock, isn't it?

Reply:
Approach with static methods is completely againts main assumption of this class. This implementation of repository was created to imitate real database until MySQL is provided in another ticket. This approach allows other developers to implement and test new tickets without waiting till WS-8 is done.  Now this "dumb" implementation of repository might be injected in each controller or service, but when real database will be created then we can easly change injected object. 

I could move this logic to constructor.

Reply:
> we can easly change injected implementation.

easier, if you don't have boiler code that you have to execute, with this implementation.

I could move this logic to constructor.

Please do.

Reply:
`easier, if you don't have boiler code that you have to execute` what do you mean? Replacing this class with real implementation will be done by add @Primary on "RealRepository" class. There is no boiler code that have to be executed.